xonsh/lib/subprocess.xsh

```diff
@@ -0,0 +1,19 @@
+"""Xonsh extension of the standard library subprocess module, using xonsh for
+subprocess calls"""
+from xonsh.tools import XonshCalledProcessError as CalledProcessError
+from xonsh.lib.os import indir
+
+
+def run(cmd, cwd=None, check=False):
+    """Drop in replacement for ``subprocess.run`` like functionality"""
+    if cwd is None:
+        cwd = '.'
+    with indir(cwd), ${...}.swap(RAISE_SUBPROC_ERROR=check):
+        p = ![@(cmd)]
+    return p
+
+
+def check_call(cmd, cwd=None):
+    """Drop in replacement for ``subprocess.check_call`` like functionality"""
+    run(cmd, cwd=cwd, check=True)
+    return 0
```

Comment:
This isn't exactly what I meant :)  I would like to see, 

```
p = run(...)
return p.returncode
```

Reply:
Ok, I was trying to mirror the extract syntax offered by check call.

Reply:
Right, so my concern is that if for some reason run returns with a non-zero return code, the current version of `check_call()` would mask us from ever knowing.